src/System.AppContext/tests/AppContext.BaseDirectory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Diagnostics;
+using Xunit;
+
+namespace System.Tests
+{
+    public partial class AppContextTests
+    {
+        [Fact]
+        [ActiveIssue(6677, PlatformID.OSX)]
+        public void DefaultMatchesGetModuleFileName()
+        {
+            Assert.Equal(GetMainModuleDirectory(), AppContext.BaseDirectory.TrimEnd(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar));
+        }
+
+        private static string GetMainModuleDirectory()
```

Comment:
This test is making a number of assumptions that will soon be invalid when we enable it for running on .NET Framework and .NET Native which we are currently working on enabling. So not only will this not be applicable to non-windows the other frameworks will also not be correct, so some further thought needs to go into how to test this functionality.